cmd/api/src/api/v2/database_wipe_integration_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build integration
+// +build integration
+
+package v2_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gofrs/uuid"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/lab"
+	v2 "github.com/specterops/bloodhound/src/api/v2"
+	"github.com/specterops/bloodhound/src/test/lab/fixtures"
+	"github.com/specterops/bloodhound/src/test/lab/harnesses"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDatabaseWipe_CollectedGraphData(t *testing.T) {
+	var (
+		harness       = harnesses.NewIntegrationTestHarness(fixtures.BHAdminApiClientFixture)
+		graphdb       = fixtures.NewGraphDBFixture()
+		computer1UUID = uuid.Must(uuid.NewV4())
+		computer1     = fixtures.NewComputerFixture(computer1UUID, "computer 1", fixtures.BasicDomainFixture)
+		computer2UUID = uuid.Must(uuid.NewV4())
+		computer2     = fixtures.NewComputerFixture(computer2UUID, "computer 2", fixtures.BasicDomainFixture)
+		computer3UUID = uuid.Must(uuid.NewV4())
+		computer3     = fixtures.NewComputerFixture(computer3UUID, "computer 3", fixtures.BasicDomainFixture)
+	)
+
+	lab.Pack(harness, graphdb)
+	lab.Pack(harness, computer1)
+	lab.Pack(harness, computer2)
+	lab.Pack(harness, computer3)
+
+	lab.NewSpec(t, harness).Run(
+		lab.TestCase("the endpoint deletes graph data", func(assert *require.Assertions, harness *lab.Harness) {
```

Comment:
i added a lot of unit test coverage to this function to cover more bases.  our integration test suite like lab doesn't support writing a test like "create some high value selectors in the db, call the wipe database endpoint, and then assert that there are no high value selectors in the db."

i tried to follow this approach, but what i was running into was a race condition in the test.  i would setup some fixtures in lab to create some data, i.e. some high value selectors, but asset group selectors depend on a FK relationship to an asset group, so I made a fixture to create the asset group first, and then a selector.

some really wonky behavior happened where the asset group record would be created in memory in gorm but wouldn't be committed to the db, so there was a race condition when trying to create the selector.  creating the selector usually threw a FK constraint violation because the asset group wasn't actually committed to the db yet.

it was too burdensome to try to get something to work for the integration test, so i just unit tested instead!